yaptide/batch/batch_methods.py

```diff
@@ -18,27 +18,37 @@
     COLLECT_BASH
 )
 from yaptide.batch.utils.utils import extract_sbatch_header, convert_dict_to_sbatch_options
-from yaptide.persistence.models import SimulationModel, ClusterModel
+from yaptide.persistence.models import KeycloakUserModel, SimulationModel, ClusterModel
 from yaptide.utils.sim_utils import write_simulation_input_files
 
 
-def submit_job(payload_dict: dict, files_dict: dict, cluster: ClusterModel) -> dict:
-    """Dummy version of submit_job"""
-    utc_time = int(datetime.utcnow().timestamp()*1e6)
-    pkey = Ed25519Key(file_obj=io.StringIO(cluster.cluster_ssh_key))
+def get_connection(user: KeycloakUserModel, cluster: ClusterModel) -> Connection:
+    """Returns connection object to cluster"""
+    pkey = RSAKey.from_private_key(io.StringIO(user.private_key))
+    pkey.load_certificate(user.cert)
+
     con = Connection(
-        host=f"{cluster.cluster_username}@{cluster.cluster_name}",
-        connect_kwargs={"pkey": pkey}
+        host=f"{user.username}@{cluster.cluster_name}",
+        connect_kwargs={
+            "pkey": pkey,
+            "allow_agent": False,
+            "look_for_keys": False
+        }
     )
+    return con
+
+
+def submit_job(payload_dict: dict, files_dict: dict, user: KeycloakUserModel, cluster: ClusterModel) -> dict:
+    """Dummy version of submit_job"""
+    utc_time = int(datetime.utcnow().timestamp()*1e6)
+
+    con = get_connection(user=user, cluster=cluster)
```

Comment:
At which point we connect to the cluster ? At getting connection or at calling `con.run` ?

Which will fail if we have invalid certificate ?

Reply:
You should handle here the case where cert is invalid